music21/expressions.py

```diff
@@ -325,6 +325,114 @@ def nextMark(self):
         return RehearsalMark(self.nextContent(), numbering=self.numbering)
 
 
+# ------------------------------------------------------------------------------
+class PedalMark(spanner.Spanner):
+    '''
+        A pedal mark spanner contains a pedaled set of notes.
+        The spanner starts at the moment of down-pedal, and
+        ends at the moment of up-pedal.  There can be any
+        number of pedal "bounces" in the middle of the spanner.
+
+        The visual form of the pedal mark can vary.  The following
+        examples use a pedal mark with one "bounce" in the middle.
+
+        Pedal marks can be lines:            |_______^________|
+        Pedal marks can be normal symbolic:  Ped.    * Ped.   *
+        Pedal marks can be altered symbolic: Ped.    Ped.     *
+        Pedal marks can be symbol and line:  Ped.____^________|
+
+        Pedal marks, whether lines, symbols, or a combination, can
+        represent the sustain pedal (Ped.), the sostenuto pedal
+        (Sost.), the soft (una corda) pedal, or (more rarely) the
+        silent (muting) pedal.
+
+        Pedal marks that are symbolic can be abbreviated: e.g. P. instead
+        of Ped., S. instead of Sost.
+
+        Pedal marks that are lines can have non-printed portions
+        (gaps) in them; these are usually started with "simile",
+        but not necessarily.
+    '''
+    validPedalTypes: tuple[str, ...] = ('sustain', 'sostenuto', 'soft', 'silent')
+    validPedalForms: tuple[str, ...] = ('line', 'symbol', 'altsymbol', 'symline')
+
+    def __init__(
+        self,
+        *spannedElements,
+        startOffset: OffsetQL|None = None,
+        **keywords
+    ):
+        super().__init__(*spannedElements, **keywords)
+        from music21 import note
+        self.fillElementTypes = [note.GeneralNote]
+
+        # startOffset is completely optional.  Can be set by parsers (such
+        # as MusicXML) that might need to get this offset back from the
+        # spanner before the start note has been added to the spanner.
+        self.startOffset: OffsetQL|None = startOffset
+
+        self.pedalType: str = 'sustain'
+        self.pedalForm: str = 'symbol'
+        self.abbreviated: bool = False
+        self.placement: str|None = None
+
+    @property
+    def pedalType(self) -> str:
```

Comment:
docs for properties (even just short ones) are nice to have.  But if Enums are used for these then there is no longer a need to check type on setter (users' IDEs should do that for them) so these can just be regular attributes.